xontrib/abbrevs.py

```diff
@@ -0,0 +1,63 @@
+"""
+Command abbreviations.
+
+This expands input words from `abbrevs` disctionary as you type.
+"""
+
+import builtins
+from prompt_toolkit.filters import IsMultiline
+from prompt_toolkit.keys import Keys
+from xonsh.platform import ptk_shell_type
+from xonsh.tools import check_for_partial_string
+
+__all__ = ()
+
+if "abbrevs" not in builtins.__xonsh__.ctx.keys():
+    builtins.__xonsh__.ctx["abbrevs"] = dict()
```

Comment:
This shouldn't be in the execution context, because it might overlap with variable names.  Rather it should be similar to the `events` or `aliases` dict which lives on the session (`builtins.__xonsh__.abbrevs`), but is linked to the `builtins.abbrevs` variable with a `DynamicAccessProxy`,  seen here https://github.com/xonsh/xonsh/blob/master/xonsh/built_ins.py#L1426

Reply:
Done. Thanks for the review.